wavpack.md

```diff
@@ -114,41 +72,47 @@ To save space and avoid redundant information in Matroska we remove data from th
-}
-```
+WavPack has a hybrid mode that splits the audio frames between lossy and correction packets.
+Adding both gives a lossless version of the original audio.
+It is possible to only store the lossy part in Matroska or both together.
+Storing only the lossy part is equivalent to the format described in (#lossless-and-lossy-storage).
+This section explains how to store all hybrid data in Matroska.
+
+Hybrid WavPack is encoded in 2 files.
+The first one has a lossy part and the second file has the correction part to reconstruct the original audio losslessly.
 
-* Block
+Each WavPack block in the correction file corresponds to a WavPack block in the lossy file with the same number of samples, that's also true for a multichannel file.
+This means that if a frame is made of 4 WavPack blocks, the correction file will have 4 WavPack blocks in the corresponding frame.
+The header of the correction WavPack block is exactly the same as in the lossy WavPack block, except for the CRC.
+
+In Matroska, the correction part is stored as an additional data available to the `Block` (see (#block-additional-mapping)).
+This way a file could be remuxed and not keep the Block Additional data and still be usable as a lossy WavPack file.
+The `Block` data of the lossy file are stored exactly the same as for lossy storage defined in (#lossless-and-lossy-storage).
+
+A `BlockAdditionMapping` **MUST** be used for hybrid WavPack `TrackEntry`'.
+
+The `BlockAddIDType` of that `BlockAdditionMapping` **MUST** be set to 1 for hybrid WavPack, corresponding to Opaque data; see (#opaque-data).
+
+The `BlockAddID` in the `BlockAdditions` **MUST** be 1, i.e., the default value, to store the correction data.
+
+#### Mono/Stereo
+
+The `BlockAdditional` data contain the following header with the "crc" field from the `WavpackHeader` of the WavPack block of the correction file
+matching the WavPack block of the lossy frame used to fill the `Block` data, followed by the data of that correction file WavPack block.
 
 ```c
 {
-  uint32_t block_samples; // # samples in this block
-  uint32_t flags;         // various flags for id and decoding
-  uint32_t crc;           // crc for actual decoded data
-  uint32_t blocksize;     // size of the data to follow
-}
-[ block data # 1 ]
-{
-  uint32_t flags;         // various flags for id and decoding
   uint32_t crc;           // crc for actual decoded data
-  uint32_t blocksize;     // size of the data to follow
 }
-[ block data # 2 ]
-{
-  uint32_t flags;         // various flags for id and decoding
-  uint32_t crc;           // crc for actual decoded data
-  uint32_t blocksize;     // size of the data to follow
-}
-[ block data # 3 ]
-...
+[ correction block data ]
 ```
 
-* BlockAdditional (level 1)
+#### Multichannel
+
+The `BlockAdditional` data contain the following header with the data from the each `WavpackHeader` of the WavPack block of the correction file
```

Comment:
`contain` > `contains`

Reply:
"data" is plural in English (and latin). If it's changed to `BlockAdditional` element, it would work. And it may be more explicit too.